processor/metricstransformprocessor/metrics_transform_processor_helper_test.go

```diff
@@ -0,0 +1,636 @@
+							AggregatedValuesSet: map[string]bool{"label2-value1": true, "label2-value2": true},
+							NewValue:            "new/label2-value",
+							AggregationType:     Sum,
+						},
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(2, []string{"label1-value1", "label2-value1"}).addTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					addInt64Point(0, 3, 2).addInt64Point(1, 1, 2).
+					build(),
+			},
+			out: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(2, []string{"label1-value1", "label2-value1"}).addTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					addInt64Point(0, 3, 2).addInt64Point(1, 1, 2).
+					build(),
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(1, []string{"label1-value1", "new/label2-value"}).
+					addInt64Point(0, 4, 2).
+					build(),
+			},
+		},
+		{
+			name: "metric_labels_aggregation_sum_distribution_insert",
+			transforms: []Transform{
+				{
+					MetricName: "metric1",
+					Action:     Insert,
+					Operations: []Operation{
+						{
+							Action:          AggregateLabels,
+							LabelSetMap:     map[string]bool{"label1": true},
+							AggregationType: Sum,
+						},
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1", "label2-value1"}).addTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					addDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					addDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					build(),
+			},
+			out: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1", "label2-value1"}).addTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					addDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					addDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					build(),
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1"}).
+					addDistributionPoints(0, 1, 8, 16, []float64{1, 2}, []int64{1, 2, 5}, 7).
+					build(),
+			},
+		},
+	}
+)
+
+// analyzeSlice returns the sum and the sumOfSquaredDeviation for this slice
+func analyzeSlice(slice []float64) (sum float64, sumOfSquaredDeviation float64) {
```

Comment:
```suggestion
func calculateSumOfSquaredDeviation(slice []float64) (sum float64, sumOfSquaredDeviation float64) {
```

Reply:
Since this function returns two values, so I thought it would be a bit misleading to have the name focus on one input, but on a second thought, the SSD is the main output so it would still make sense. Updated!

Reply:
Yea I saw` sum` as basically a byproduct of the SSD calc